packages/blade-mcp/src/utils/analyticsUtils.ts

```diff
@@ -1,12 +1,28 @@
 import os from 'os';
 import crypto from 'crypto';
+import { basename } from 'path';
 import * as Sentry from '@sentry/node';
 import { Analytics } from '@segment/analytics-node';
 import { getPackageJSONVersion } from './generalUtils.js';
 import { getUserName } from './getUserName.js';
 
 let cachedMachineId: string | null = null;
 
+type MCPSSeAnalyticsContext = {
+  protocol: 'http' | 'stdio';
+};
+
+// Context to track if the current call is from MCP SSE
+let mcpSseAnalyticsContext: MCPSSeAnalyticsContext = {
+  protocol: 'stdio',
+};
+
+const setMcpSseAnalyticsContext = ({ protocol }: MCPSSeAnalyticsContext): void => {
```

Comment:
https://github.com/razorpay/frontend-core-tools/pull/188

Whatever changes required in BE, let's do that too. Should be small.